python/lsst/daf/butler/butler.py

```diff
@@ -546,3 +551,141 @@ def remove(self, datasetRefOrType, dataId=None, *, delete=True, remember=True, *
+            composite, all component of that `DatasetType` will also be
+            ignored.
+
+        Raises
+        ------
+        ButlerValidationError
+            Raised if there is some inconsistency with how this Butler
+            is configured.
+        """
+        if datasetTypeNames:
+            entities = [self.registry.getDatasetType(name) for name in datasetTypeNames]
+        else:
+            entities = list(self.registry.getAllDatasetTypes())
+
+        # filter out anything from the ignore list
+        if ignore:
+            ignore = set(ignore)
+            entities = [e for e in entities if e.name not in ignore and e.nameAndComponent()[0] not in ignore]
+        else:
+            ignore = set()
+
+        # Find all the registered instruments
+        instruments = set()
+        if not self.registry.limited:
+            instrumentEntries = self.registry.findDimensionEntries("Instrument")
+            instruments = {e["instrument"] for e in instrumentEntries}
+
+        # For each datasetType that has an Instrument dimension, create
+        # a DatasetRef for each defined instrument
+        datasetRefs = []
+
+        for datasetType in entities:
+            if "Instrument" in datasetType.dimensions:
+                for instrument in instruments:
+                    datasetRef = DatasetRef(datasetType, {"instrument": instrument})
+                    datasetRefs.append(datasetRef)
+
+        entities.extend(datasetRefs)
+
+        datastoreErrorStr = None
+        try:
+            self.datastore.validateConfiguration(entities, logFailures=logFailures)
+        except ValidationError as e:
+            datastoreErrorStr = str(e)
+
+        # Also check that the LookupKeys used by the datastores match
+        # registry and storage class definitions
+        keys = self.datastore.getLookupKeys()
+
+        # Dummy StorageClass to use for Dimensions validation
+        dummyStorageClass = self.storageClasses.getStorageClass("StructuredDataDict")
+
+        failedNames = set()
+        failedDataId = set()
+        failedValues = set()
+        for key in keys:
+            datasetType = None
+            if key.name is not None:
+                if key.name in ignore:
+                    continue
+
+                # skip if specific datasetType names were requested and this
+                # name does not match
+                if datasetTypeNames and key.name not in datasetTypeNames:
+                    continue
+
+                # See if it is a StorageClass or a DatasetType
+                if key.name in self.storageClasses:
+                    pass
+                else:
+                    try:
+                        datasetType = self.registry.getDatasetType(key.name)
+                    except KeyError:
+                        if logFailures:
+                            log.fatal(f"Key '{key}' does not correspond to a DatasetType or StorageClass")
```

Comment:
Does not flake8 warn that you are supposed to use `%`-style formatting with logging? Or is it confused by f-strings?